.github/workflows/ci.yml

```diff
@@ -58,12 +58,27 @@ jobs:
           aws-region: us-east-1
       - name: Pull the grand-challenge container images
         run: |
-          aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/diag-nijmegen/grand-challenge/web
-          docker pull public.ecr.aws/diag-nijmegen/grand-challenge/web:latest
-          docker logout public.ecr.aws/diag-nijmegen/grand-challenge/web
-          aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/diag-nijmegen/grand-challenge/http
-          docker pull public.ecr.aws/diag-nijmegen/grand-challenge/http:latest
-          docker logout public.ecr.aws/diag-nijmegen/grand-challenge/http
+          retry() {
+            local -r -i max_attempts="$1"; shift
+            local -i attempt_num=1
+            until eval "$@"
+            do
+              command_return=$?
+              if ((attempt_num==max_attempts))
+              then
+                return $command_return
+              else
+                echo "Retrying..."
+                sleep $((attempt_num++ * 10))
+              fi
+            done
+          }
```

Comment:
I think this is probably correct, but I wonder why we are not using one of the solutions that are available right there - either on the commandline or on the marketplace for GitHub Actions? I found a few from a quick googling in the marketplace - the first one would probably already work:

- https://github.com/marketplace/actions/shell-command-retry-action
- https://github.com/marketplace/actions/retry-action#retry-command
- https://github.com/marketplace/actions/retry-step

My favorite solution however, would be to just use `retry` om thje console:

~~~~~~~~~~~~
root@2e7d0e40a0c9:/# apt install retry
[...]
root@2e7d0e40a0c9:/# retry --help


NAME
  retry - Repeat command until a criteria is met, usually success.

SYNOPSIS
  retry [-v] [-h] [-u until] [-w while] command ...

DESCRIPTION

  The tool repeats the given command until the command is successful,
  backing off with a configurable delay between each attempt.

[...]
~~~~~~~~~~~~

Reply:
PR #130 contains my fix for this...